pypdf/generic/_link.py

```diff
@@ -100,7 +100,8 @@ def _build_link(indirect_object: IndirectObject, page: "PageObject") -> Optional
         if action.get("/S") != "/GoTo":
             return None
 
-        return _create_link(action["/D"], src)
+        if "/D" in action:
```

Comment:
To avoid one lookup, we should probably return `None` if the condition is unmet.